addons/pos_self_order/models/res_config_settings.py

```diff
@@ -52,12 +52,21 @@ def _onchange_pos_self_order_kiosk(self):
             self.is_kiosk_mode = True
             self.pos_module_pos_restaurant = False
             self.pos_self_ordering_pay_after = "each"
+            for pm in self.pos_payment_method_ids.filtered(lambda x: x.is_cash_count):
+                self.pos_payment_method_ids = [(3, pm.id)]
```

Comment:
do not use `unlink` in an `onchange` https://www.odoo.com/documentation/17.0/developer/reference/backend/orm.html#module-odoo.api:~:text=will%20be%20ignored-,Danger,-Since

Reply:
Try
`self.pos_payment_method_ids -= pm.id`